app/controllers/api_controller/generic.rb

```diff
@@ -5,6 +5,7 @@ module Generic
     #
 
     def show_generic(type)
+      validate_api_action
```

Comment:
Just as a note/question.  Not sure of the background behind the show vs list identifier in miq product feature, but if there are really 2 types of entitlements, we might need to allow passing a method to validate_api_action (i.e. show vs list) based on if @req[:c_id] exists.